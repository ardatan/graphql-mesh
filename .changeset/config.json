{
  "$schema": "https://unpkg.com/@changesets/config@2.1.0/schema.json",
  "commit": false,
  "linked": [],
  "access": "restricted",
  "baseBranch": "master",
  "updateInternalDependencies": "patch",
  "ignore": ["website"],
  "changelog": ["@changesets/changelog-github", { "repo": "urigo/graphql-mesh" }],
  "snapshot": {
    "useCalculatedVersion": true,
    "prereleaseTemplate": "{tag}-{datetime}-{commit}"
  }
}
