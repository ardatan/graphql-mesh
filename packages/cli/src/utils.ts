export const stripJSONComments = (data: string) => {
  const re = /\/\/(.*)/g;
  const re2 = /\/\*(.*)\*\//g;
  return data.replace(re, '').replace(re2, '');
};
