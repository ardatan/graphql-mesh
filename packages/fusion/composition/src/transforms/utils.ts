export class TransformValidationError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'TransformValidationError';
  }
}
