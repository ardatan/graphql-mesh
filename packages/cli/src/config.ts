import { ConfigProcessOptions, processConfig } from '@graphql-mesh/config';
import { jsonSchema, YamlConfig } from '@graphql-mesh/types';
import { defaultImportFn, loadYaml } from '@graphql-mesh/utils';
import Ajv from 'ajv';
import { cosmiconfig, defaultLoaders } from 'cosmiconfig';
import path from 'path';

export function validateConfig(config: any): asserts config is YamlConfig.Config {
  const ajv = new Ajv({
    strict: false,
  } as any);
  jsonSchema.$schema = undefined;
  const isValid = ajv.validate(jsonSchema, config);
  if (!isValid) {
    console.warn(`Configuration is not valid:\n${ajv.errorsText()}`);
  }
}

export async function findAndParseConfig(options?: ConfigProcessOptions) {
  const { configName = 'mesh', dir: configDir = '', ...restOptions } = options || {};
  const dir = path.isAbsolute(configDir) ? configDir : path.join(process.cwd(), configDir);
  const explorer = cosmiconfig(configName, {
    loaders: {
      '.json': customLoader('json', options?.importFn),
      '.yaml': customLoader('yaml', options?.importFn),
      '.yml': customLoader('yaml', options?.importFn),
      '.js': customLoader('js', options?.importFn),
      '.ts': customLoader('js', options?.importFn),
      noExt: customLoader('yaml', options?.importFn),
    },
  });
  const results = await explorer.search(dir);

  if (!results) {
    throw new Error(`No ${configName} config file found in "${dir}"!`);
  }

  const config = results.config;
  validateConfig(config);
  return processConfig(config, { dir, ...restOptions });
}

function customLoader(ext: 'json' | 'yaml' | 'js', importFn = defaultImportFn) {
  function loader(filepath: string, content: string) {
    if (process.env) {
      content = content.replace(/\$\{(.*?)\}/g, (_, variable) => {
        let varName = variable;
        let defaultValue = '';

        if (variable.includes(':')) {
          const spl = variable.split(':');
          varName = spl.shift();
          defaultValue = spl.join(':');
        }

        return process.env[varName] || defaultValue;
      });
    }

    if (ext === 'json') {
      return defaultLoaders['.json'](filepath, content);
    }

    if (ext === 'yaml') {
      return loadYaml(filepath, content);
    }

    if (ext === 'js') {
      return importFn(filepath);
    }
  }

  return loader;
}
