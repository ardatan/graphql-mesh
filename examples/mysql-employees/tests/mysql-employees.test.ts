import { basename, join } from 'path';
import { findAndParseConfig } from '@graphql-mesh/cli';
import { ProcessedConfig } from '@graphql-mesh/config';
import { getMesh, MeshInstance } from '@graphql-mesh/runtime';
import { printSchemaWithDirectives } from '@graphql-tools/utils';

jest.setTimeout(30000);

describe('MySQL Employees', () => {
  let mesh: MeshInstance;
  let config: ProcessedConfig;
  beforeAll(async () => {
    config = await findAndParseConfig({
      dir: join(__dirname, '..'),
    });
    mesh = await getMesh(config);
  });
  it('should generate correct schema', async () => {
    expect(printSchemaWithDirectives(mesh.schema)).toMatchSnapshot('schema');
  });
  it('should give correct response for example queries', () =>
    Promise.all(
      config.documents.map(async operation => {
        const result = await mesh.execute(operation.document!);
        expect(result).toMatchSnapshot(
          basename(operation.location || 'anonymous') + '-query-result',
        );
      }),
    ));
  afterAll(() => {
    mesh?.destroy();
  });
});
