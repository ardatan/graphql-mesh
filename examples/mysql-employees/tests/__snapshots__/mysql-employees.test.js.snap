// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySQL Employees should generate correct schema: mysql-employees-schema 1`] = `
"\\"\\"\\"
A date string, such as 2007-12-03, compliant with the \`full-date\` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
\\"\\"\\"
scalar Date

type Mutation {
  delete_current_dept_emp(where: current_dept_emp_WhereInput): Boolean
  delete_departments(where: departments_WhereInput): Boolean
  delete_dept_emp(where: dept_emp_WhereInput): Boolean
  delete_dept_emp_latest_date(where: dept_emp_latest_date_WhereInput): Boolean
  delete_dept_manager(where: dept_manager_WhereInput): Boolean
  delete_employees(where: employees_WhereInput): Boolean
  delete_salaries(where: salaries_WhereInput): Boolean
  delete_titles(where: titles_WhereInput): Boolean
  delete_v_full_departments(where: v_full_departments_WhereInput): Boolean
  delete_v_full_employees(where: v_full_employees_WhereInput): Boolean
  insert_current_dept_emp(current_dept_emp: current_dept_emp_InsertInput!): current_dept_emp
  insert_departments(departments: departments_InsertInput!): departments
  insert_dept_emp(dept_emp: dept_emp_InsertInput!): dept_emp
  insert_dept_emp_latest_date(dept_emp_latest_date: dept_emp_latest_date_InsertInput!): dept_emp_latest_date
  insert_dept_manager(dept_manager: dept_manager_InsertInput!): dept_manager
  insert_employees(employees: employees_InsertInput!): employees
  insert_salaries(salaries: salaries_InsertInput!): salaries
  insert_titles(titles: titles_InsertInput!): titles
  insert_v_full_departments(v_full_departments: v_full_departments_InsertInput!): v_full_departments
  insert_v_full_employees(v_full_employees: v_full_employees_InsertInput!): v_full_employees
  update_current_dept_emp(current_dept_emp: current_dept_emp_UpdateInput!, where: current_dept_emp_WhereInput): current_dept_emp
  update_departments(departments: departments_UpdateInput!, where: departments_WhereInput): departments
  update_dept_emp(dept_emp: dept_emp_UpdateInput!, where: dept_emp_WhereInput): dept_emp
  update_dept_emp_latest_date(dept_emp_latest_date: dept_emp_latest_date_UpdateInput!, where: dept_emp_latest_date_WhereInput): dept_emp_latest_date
  update_dept_manager(dept_manager: dept_manager_UpdateInput!, where: dept_manager_WhereInput): dept_manager
  update_employees(employees: employees_UpdateInput!, where: employees_WhereInput): employees
  update_salaries(salaries: salaries_UpdateInput!, where: salaries_WhereInput): salaries
  update_titles(titles: titles_UpdateInput!, where: titles_WhereInput): titles
  update_v_full_departments(v_full_departments: v_full_departments_UpdateInput!, where: v_full_departments_WhereInput): v_full_departments
  update_v_full_employees(v_full_employees: v_full_employees_UpdateInput!, where: v_full_employees_WhereInput): v_full_employees
}

enum OrderBy {
  asc
  desc
}

type Query {
  count_current_dept_emp(where: current_dept_emp_WhereInput): Int
  count_departments(where: departments_WhereInput): Int
  count_dept_emp(where: dept_emp_WhereInput): Int
  count_dept_emp_latest_date(where: dept_emp_latest_date_WhereInput): Int
  count_dept_manager(where: dept_manager_WhereInput): Int
  count_employees(where: employees_WhereInput): Int
  count_salaries(where: salaries_WhereInput): Int
  count_titles(where: titles_WhereInput): Int
  count_v_full_departments(where: v_full_departments_WhereInput): Int
  count_v_full_employees(where: v_full_employees_WhereInput): Int
  current_dept_emp(limit: Int, offset: Int, orderBy: current_dept_emp_OrderByInput, where: current_dept_emp_WhereInput): [current_dept_emp]
  departments(limit: Int, offset: Int, orderBy: departments_OrderByInput, where: departments_WhereInput): [departments]
  dept_emp(limit: Int, offset: Int, orderBy: dept_emp_OrderByInput, where: dept_emp_WhereInput): [dept_emp]
  dept_emp_latest_date(limit: Int, offset: Int, orderBy: dept_emp_latest_date_OrderByInput, where: dept_emp_latest_date_WhereInput): [dept_emp_latest_date]
  dept_manager(limit: Int, offset: Int, orderBy: dept_manager_OrderByInput, where: dept_manager_WhereInput): [dept_manager]
  employees(limit: Int, offset: Int, orderBy: employees_OrderByInput, where: employees_WhereInput): [employees]
  salaries(limit: Int, offset: Int, orderBy: salaries_OrderByInput, where: salaries_WhereInput): [salaries]
  titles(limit: Int, offset: Int, orderBy: titles_OrderByInput, where: titles_WhereInput): [titles]
  v_full_departments(limit: Int, offset: Int, orderBy: v_full_departments_OrderByInput, where: v_full_departments_WhereInput): [v_full_departments]
  v_full_employees(limit: Int, offset: Int, orderBy: v_full_employees_OrderByInput, where: v_full_employees_WhereInput): [v_full_employees]
}

\\"\\"\\"VIEW\\"\\"\\"
type current_dept_emp {
  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"VIEW\\"\\"\\"
input current_dept_emp_InsertInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"VIEW\\"\\"\\"
input current_dept_emp_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  from_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  to_date: OrderBy
}

\\"\\"\\"VIEW\\"\\"\\"
input current_dept_emp_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"VIEW\\"\\"\\"
input current_dept_emp_WhereInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  from_date: String

  \\"\\"\\"\\"\\"\\"
  to_date: String
}

\\"\\"\\"\\"\\"\\"
type departments {
  \\"\\"\\"\\"\\"\\"
  dept_name: String!

  \\"\\"\\"\\"\\"\\"
  dept_no: String!
}

\\"\\"\\"\\"\\"\\"
input departments_InsertInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: String!

  \\"\\"\\"\\"\\"\\"
  dept_no: String!
}

\\"\\"\\"\\"\\"\\"
input departments_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: OrderBy

  \\"\\"\\"\\"\\"\\"
  dept_no: OrderBy
}

\\"\\"\\"\\"\\"\\"
input departments_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: String

  \\"\\"\\"\\"\\"\\"
  dept_no: String
}

\\"\\"\\"\\"\\"\\"
input departments_WhereInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: String

  \\"\\"\\"\\"\\"\\"
  dept_no: String
}

\\"\\"\\"\\"\\"\\"
type dept_emp {
  departments(limit: Int, offset: Int, orderBy: departments_OrderByInput, where: departments_WhereInput): [departments]

  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!
  employees(limit: Int, offset: Int, orderBy: employees_OrderByInput, where: employees_WhereInput): [employees]

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  to_date: Date!
}

\\"\\"\\"\\"\\"\\"
input dept_emp_InsertInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  to_date: Date!
}

\\"\\"\\"\\"\\"\\"
input dept_emp_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  from_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  to_date: OrderBy
}

\\"\\"\\"\\"\\"\\"
input dept_emp_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"\\"\\"\\"
input dept_emp_WhereInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  from_date: String

  \\"\\"\\"\\"\\"\\"
  to_date: String
}

\\"\\"\\"VIEW\\"\\"\\"
type dept_emp_latest_date {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"VIEW\\"\\"\\"
input dept_emp_latest_date_InsertInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"VIEW\\"\\"\\"
input dept_emp_latest_date_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  from_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  to_date: OrderBy
}

\\"\\"\\"VIEW\\"\\"\\"
input dept_emp_latest_date_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"VIEW\\"\\"\\"
input dept_emp_latest_date_WhereInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  from_date: String

  \\"\\"\\"\\"\\"\\"
  to_date: String
}

\\"\\"\\"\\"\\"\\"
type dept_manager {
  departments(limit: Int, offset: Int, orderBy: departments_OrderByInput, where: departments_WhereInput): [departments]

  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!
  employees(limit: Int, offset: Int, orderBy: employees_OrderByInput, where: employees_WhereInput): [employees]

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  to_date: Date!
}

\\"\\"\\"\\"\\"\\"
input dept_manager_InsertInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  to_date: Date!
}

\\"\\"\\"\\"\\"\\"
input dept_manager_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  from_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  to_date: OrderBy
}

\\"\\"\\"\\"\\"\\"
input dept_manager_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"\\"\\"\\"
input dept_manager_WhereInput {
  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  from_date: String

  \\"\\"\\"\\"\\"\\"
  to_date: String
}

\\"\\"\\"\\"\\"\\"
type employees {
  \\"\\"\\"\\"\\"\\"
  birth_date: Date!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  first_name: String!

  \\"\\"\\"\\"\\"\\"
  gender: employees_gender!

  \\"\\"\\"\\"\\"\\"
  hire_date: Date!

  \\"\\"\\"\\"\\"\\"
  last_name: String!
}

\\"\\"\\"\\"\\"\\"
input employees_InsertInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: Date!

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  first_name: String!

  \\"\\"\\"\\"\\"\\"
  gender: employees_gender!

  \\"\\"\\"\\"\\"\\"
  hire_date: Date!

  \\"\\"\\"\\"\\"\\"
  last_name: String!
}

\\"\\"\\"\\"\\"\\"
input employees_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  first_name: OrderBy

  \\"\\"\\"\\"\\"\\"
  gender: OrderBy

  \\"\\"\\"\\"\\"\\"
  hire_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  last_name: OrderBy
}

\\"\\"\\"\\"\\"\\"
input employees_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: Date

  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  first_name: String

  \\"\\"\\"\\"\\"\\"
  gender: employees_gender

  \\"\\"\\"\\"\\"\\"
  hire_date: Date

  \\"\\"\\"\\"\\"\\"
  last_name: String
}

\\"\\"\\"\\"\\"\\"
input employees_WhereInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: String

  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  first_name: String

  \\"\\"\\"\\"\\"\\"
  gender: String

  \\"\\"\\"\\"\\"\\"
  hire_date: String

  \\"\\"\\"\\"\\"\\"
  last_name: String
}

enum employees_gender {
  F
  M
}

\\"\\"\\"\\"\\"\\"
type salaries {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int!
  employees(limit: Int, offset: Int, orderBy: employees_OrderByInput, where: employees_WhereInput): [employees]

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  salary: Int!

  \\"\\"\\"\\"\\"\\"
  to_date: Date!
}

\\"\\"\\"\\"\\"\\"
input salaries_InsertInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  salary: Int!

  \\"\\"\\"\\"\\"\\"
  to_date: Date!
}

\\"\\"\\"\\"\\"\\"
input salaries_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  from_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  salary: OrderBy

  \\"\\"\\"\\"\\"\\"
  to_date: OrderBy
}

\\"\\"\\"\\"\\"\\"
input salaries_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  salary: Int

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"\\"\\"\\"
input salaries_WhereInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  from_date: String

  \\"\\"\\"\\"\\"\\"
  salary: String

  \\"\\"\\"\\"\\"\\"
  to_date: String
}

\\"\\"\\"\\"\\"\\"
type titles {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int!
  employees(limit: Int, offset: Int, orderBy: employees_OrderByInput, where: employees_WhereInput): [employees]

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  title: String!

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"\\"\\"\\"
input titles_InsertInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  from_date: Date!

  \\"\\"\\"\\"\\"\\"
  title: String!

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"\\"\\"\\"
input titles_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  from_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  title: OrderBy

  \\"\\"\\"\\"\\"\\"
  to_date: OrderBy
}

\\"\\"\\"\\"\\"\\"
input titles_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  from_date: Date

  \\"\\"\\"\\"\\"\\"
  title: String

  \\"\\"\\"\\"\\"\\"
  to_date: Date
}

\\"\\"\\"\\"\\"\\"
input titles_WhereInput {
  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  from_date: String

  \\"\\"\\"\\"\\"\\"
  title: String

  \\"\\"\\"\\"\\"\\"
  to_date: String
}

\\"\\"\\"VIEW\\"\\"\\"
type v_full_departments {
  \\"\\"\\"\\"\\"\\"
  dept_name: String!

  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  manager: String
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_departments_InsertInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: String!

  \\"\\"\\"\\"\\"\\"
  dept_no: String!

  \\"\\"\\"\\"\\"\\"
  manager: String
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_departments_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: OrderBy

  \\"\\"\\"\\"\\"\\"
  dept_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  manager: OrderBy
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_departments_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: String

  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  manager: String
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_departments_WhereInput {
  \\"\\"\\"\\"\\"\\"
  dept_name: String

  \\"\\"\\"\\"\\"\\"
  dept_no: String

  \\"\\"\\"\\"\\"\\"
  manager: String
}

\\"\\"\\"VIEW\\"\\"\\"
type v_full_employees {
  \\"\\"\\"\\"\\"\\"
  birth_date: Date!

  \\"\\"\\"\\"\\"\\"
  department: String

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  first_name: String!

  \\"\\"\\"\\"\\"\\"
  gender: v_full_employees_gender!

  \\"\\"\\"\\"\\"\\"
  hire_date: Date!

  \\"\\"\\"\\"\\"\\"
  last_name: String!
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_employees_InsertInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: Date!

  \\"\\"\\"\\"\\"\\"
  department: String

  \\"\\"\\"\\"\\"\\"
  emp_no: Int!

  \\"\\"\\"\\"\\"\\"
  first_name: String!

  \\"\\"\\"\\"\\"\\"
  gender: v_full_employees_gender!

  \\"\\"\\"\\"\\"\\"
  hire_date: Date!

  \\"\\"\\"\\"\\"\\"
  last_name: String!
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_employees_OrderByInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  department: OrderBy

  \\"\\"\\"\\"\\"\\"
  emp_no: OrderBy

  \\"\\"\\"\\"\\"\\"
  first_name: OrderBy

  \\"\\"\\"\\"\\"\\"
  gender: OrderBy

  \\"\\"\\"\\"\\"\\"
  hire_date: OrderBy

  \\"\\"\\"\\"\\"\\"
  last_name: OrderBy
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_employees_UpdateInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: Date

  \\"\\"\\"\\"\\"\\"
  department: String

  \\"\\"\\"\\"\\"\\"
  emp_no: Int

  \\"\\"\\"\\"\\"\\"
  first_name: String

  \\"\\"\\"\\"\\"\\"
  gender: v_full_employees_gender

  \\"\\"\\"\\"\\"\\"
  hire_date: Date

  \\"\\"\\"\\"\\"\\"
  last_name: String
}

\\"\\"\\"VIEW\\"\\"\\"
input v_full_employees_WhereInput {
  \\"\\"\\"\\"\\"\\"
  birth_date: String

  \\"\\"\\"\\"\\"\\"
  department: String

  \\"\\"\\"\\"\\"\\"
  emp_no: String

  \\"\\"\\"\\"\\"\\"
  first_name: String

  \\"\\"\\"\\"\\"\\"
  gender: String

  \\"\\"\\"\\"\\"\\"
  hire_date: String

  \\"\\"\\"\\"\\"\\"
  last_name: String
}

enum v_full_employees_gender {
  F
  M
}"
`;

exports[`MySQL Employees should give correct response for example queries: example-query.graphql-query-result 1`] = `
Object {
  "data": Object {
    "employees": Array [
      Object {
        "__typename": "employees",
        "birth_date": "1953-09-02",
        "emp_no": 10001,
        "first_name": "Georgi",
        "gender": "M",
        "hire_date": "1986-06-26",
        "last_name": "Facello",
      },
      Object {
        "__typename": "employees",
        "birth_date": "1964-06-02",
        "emp_no": 10002,
        "first_name": "Bezalel",
        "gender": "F",
        "hire_date": "1985-11-21",
        "last_name": "Simmel",
      },
      Object {
        "__typename": "employees",
        "birth_date": "1959-12-03",
        "emp_no": 10003,
        "first_name": "Parto",
        "gender": "M",
        "hire_date": "1986-08-28",
        "last_name": "Bamford",
      },
      Object {
        "__typename": "employees",
        "birth_date": "1954-05-01",
        "emp_no": 10004,
        "first_name": "Chirstian",
        "gender": "M",
        "hire_date": "1986-12-01",
        "last_name": "Koblick",
      },
      Object {
        "__typename": "employees",
        "birth_date": "1955-01-21",
        "emp_no": 10005,
        "first_name": "Kyoichi",
        "gender": "M",
        "hire_date": "1989-09-12",
        "last_name": "Maliniak",
      },
    ],
  },
}
`;
